lib/utils/background_push.dart

```diff
@@ -50,16 +50,15 @@ class BackgroundPush {
   final FlutterLocalNotificationsPlugin _flutterLocalNotificationsPlugin =
       FlutterLocalNotificationsPlugin();
   Client client;
-  BuildContext? context;
-  GlobalKey<NavigatorState>? router;
+  GoRouter? router;
```

Comment:
Why we need to add this here?

Reply:
IMO: 
- In this class, we need a `BuildContext` to get current `location`.
- We don't have to pass the Gouter into the class. We only need a `GlobalKey<NavigatorState> ` from `GoRouterDelegate`
- I updated it, it's very clearly pls check it again